core/server/api/v2/utils/serializers/input/utils/url.js

```diff
@@ -0,0 +1,66 @@
+const {absoluteToRelative, getBlogUrl} = require('../../../../../../services/url/utils');
+const url = require('url');
+
+function handleImageUrl(imageUrl) {
+    let blogUrl = getBlogUrl();
+    if (url.parse(imageUrl).host === url.parse(blogUrl).host) {
```

Comment:
Do we need to be careful with subdirectories here? Having some tests that demonstrate expected input/output of serialisers would be useful.

Reply:
> Do we need to be careful with subdirectories here?

Hmm yeah you are right, I think ideally we should as definitely same domain/host doesn't guarantee absolute url was generated by ghost and not maybe a storage adapter etc. Ques is if `domain/subdir` is enough or should we check something like `domain/subdir/content/images/` to be sure absolute url was generated by ghost?

Reply:
I think some tests would help with that question. Specifics I'm thinking about:

configured url: https://mysite.com/
image url: https://mysite.com/mycustomstorage/images/image.jpg

^ should be fine with the current code I think because the relative path will still be correct.

configured url: https://mysite.com/blog/
image url: https://mysite.com/blog/content/images/image.jpg
image url: https://mysite.com/mycustomstorage/images/image.jpg
image url: https://mysite.com/blog/mycustomstorage/images/image.jpg

^ will depend how the relative->absolute generation works. If it's expecting URLs relative to the _ghost_ root then the second will fail, however if it's expecting URLs relative to the domain root then the existing code should still be fine but some tests should prove that

Reply:
👍 Yeah sounds good, will add test cases to cover this.